File: src/my_ude_py/scripts/check_client.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from ude_interface.srv import OddEvenCheck


class CheckClientNode(Node):
    def __init__(self):
        super().__init__("check_client")
        self.get_logger().info("Check Client Node has been started")
        self.client = self.create_client(OddEvenCheck, 'odd_even_check')
        self.req = OddEvenCheck.Request()

    def send_request(self, num):
        self.req.number = int(num)
        self.client.wait_for_service()
        self.future = self.client.call_async(self.req)
        rclpy.spin_until_future_complete(self, self.future)

        self.result = self.future.result()
        return self.result 


def main(args=None):
    rclpy.init()
    client_node = CheckClientNode()
    try:
        user_input = input("Enter an integer: ")
        res = client_node.send_request(user_input)
        print("Server returned: " + res.decision)
    except KeyboardInterrupt:
        print("Terminating Node...")
        client_node.destroy_node()


if __name__ == "__main__":
    main()

File: src/my_ude_cpp/src/check_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "ude_interface/srv/odd_even_check.hpp"

#include <iostream>

class CheckClientNode : public rclcpp::Node 
{
public:
    CheckClientNode() : Node("check_client")
    {
        threads_.push_back(std::thread(std::bind(&CheckClientNode::callOddEvenCheckService, this)));
    	RCLCPP_INFO(this->get_logger(), "Node has been started.");
    }

private:

    void callOddEvenCheckService()
    {
        auto client = this->create_client<ude_interface::srv::OddEvenCheck>("odd_even_check");
        while (!client->wait_for_service(std::chrono::seconds(1)))
        {
            RCLCPP_WARN(this->get_logger(), "Waiting for the server to be up...");
        }

        auto request = std::make_shared<ude_interface::srv::OddEvenCheck::Request>();
        std::cout << "Please type a number to check if it is Odd or Even: ";
	    std::cin >> request->number;

        auto future = client->async_send_request(request);

        try
        {
            auto response = future.get();
            RCLCPP_INFO(this->get_logger(), "the checked number is %d and the answer is %s", int(request->number), response->decision.c_str());
        }
        catch (const std::exception &e)
        {
            RCLCPP_ERROR(this->get_logger(), "Service call failed");
        }
    }

    std::vector<std::thread> threads_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CheckClientNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}